addons/portal/static/src/js/portal_name_and_signature.js

```diff
@@ -0,0 +1,582 @@
+     *  the signer.
+     * @param {string} [options.mode='draw'] - @see this.setMode
+     * @param {string} [options.signatureType='signature'] - The type of
+     *  signature used in 'auto' mode. Can be one of the following values:
+     *  - 'signature': it will adapt the characters width to fit the whole
+     *      text in the image.
+     *  - 'initial': it will adapt the space between characters to fill
+     *      the image with the text. The text will be the first letter of
+     *      every word in the name, separated by dots.
+     */
+    init: function (parent, options) {
+        this._super.apply(this, arguments);
+        options = options || {};
+        this.htmlId = _.uniqueId();
+        this.defaultName = options.defaultName || '';
+        this.signatureRatio = options.signatureRatio || 3.0;
+        this.signatureType = options.signatureType || 'signature';
+        this.signMode = options.mode || 'draw';
+        this.currentFont = 0;
+        this.drawTimeout = null;
+        this.drawPreviewTimeout = null;
+    },
+    /**
+     * Loads the fonts.
+     *
+     * @override
+     */
+    willStart: function () {
+        var self = this;
+        return $.when(
+            this._super.apply(this, arguments),
+            this._rpc({route: '/portal/sign/get_fonts'}).then(function (data) {
+                self.fonts = data;
+            })
+        );
+    },
+    /**
+     * Finds the DOM elements, initializes the signature area,
+     * and focus the name field.
+     *
+     * @override
+     */
+    start: function () {
+        var self = this;
+        // signature and name input
+        this.$signatureGroup = this.$('.o_portal_sign_signature_group');
+        this.$signatureField = this.$('.o_portal_sign_signature');
+        this.$nameInput = this.$('.o_portal_sign_name_input');
+
+        // mode selection buttons
+        this.$drawButton = this.$('a.o_portal_sign_draw_button');
+        this.$autoButton = this.$('a.o_portal_sign_auto_button');
+        this.$loadButton = this.$('a.o_portal_sign_load_button');
+
+        // mode: draw
+        this.$drawClear = this.$('.o_portal_sign_draw_clear');
+
+        // mode: auto
+        this.$autoSelectStyle = this.$('.o_portal_sign_auto_select_style');
+        this.$autoFontSelection = this.$('.o_portal_sign_auto_font_selection');
+        this.$autoFontList = this.$('.o_portal_sign_auto_font_list');
+        for (var i in this.fonts) {
+            var $img = $('<img/>').addClass('img-fluid');
+            var $a = $('<a/>').addClass('btn p-0').append($img).data('fontNb', i);
+            this.$autoFontList.append($a);
+        }
+
+        // mode: load
+        this.$loadFile = this.$('.o_portal_sign_load_file');
+        this.$loadInvalid = this.$('.o_portal_sign_load_invalid');
+
+        // Resize the signature area if it is resized
+        $(window).on('resize.o_portal_sign_name_and_signature', _.debounce(function () {
```

Comment:
The event should be unbinded in `destroy` (`.off('.o_portal_sign_name_and_signature')`)

Reply:
ok